packages/nodes-base/nodes/N8nApi/CredentialDescription.ts

```diff
@@ -0,0 +1,160 @@
+import { INodeProperties } from 'n8n-workflow';
+import { parseAndSetBodyJson } from './GenericFunctions';
+
+export const credentialOperations: INodeProperties[] = [
+	{
+		displayName: 'Operation',
+		name: 'operation',
+		type: 'options',
+		noDataExpression: true,
+		default: 'create',
+		displayOptions: {
+			show: {
+				resource: ['credential'],
+			},
+		},
+		options: [
+			{
+				name: 'Create',
+				value: 'create',
+				action: 'Create a credential',
+				routing: {
+					request: {
+						method: 'POST',
+						url: '/credentials',
+					},
+				},
+			},
+			{
+				name: 'Delete',
+				value: 'delete',
+				action: 'Delete a credential',
+				routing: {
+					request: {
+						method: 'DELETE',
+						url: '=/credentials/{{ $parameter.credentialId }}',
+					},
+				},
+			},
+			{
+				name: 'Get Schema',
+				value: 'getSchema',
+				action: 'Get credential data schema for type',
+				routing: {
+					request: {
+						method: 'GET',
+						url: '=/credentials/schema/{{ $parameter.credentialTypeName }}',
+					},
+				},
+			},
+		],
+	},
+];
+
+const createOperation: INodeProperties[] = [
+	{
+		displayName: 'Name',
+		name: 'name',
+		type: 'string',
+		default: '',
+		required: true,
+		displayOptions: {
+			show: {
+				resource: ['credential'],
+				operation: ['create'],
+			},
+		},
+		routing: {
+			request: {
+				body: {
+					name: '={{ $value }}',
+				},
+			},
+		},
+		description: 'Name of the new credential',
+	},
+	{
+		displayName: 'Credential Type Name',
+		name: 'credentialTypeName',
+		type: 'string',
+		default: '',
+		required: true,
+		displayOptions: {
+			show: {
+				resource: ['credential'],
+				operation: ['create'],
+			},
+		},
+		routing: {
+			request: {
+				body: {
+					type: '={{ $value }}',
+				},
+			},
+		},
+		description: 'The name for the type of this credential',
+	},
+	{
+		displayName: 'Data',
+		name: 'data',
+		type: 'json',
+		default: '{}',
+		required: true,
+		typeOptions: {
+			alwaysOpenEditWindow: true,
+		},
+		displayOptions: {
+			show: {
+				resource: ['credential'],
+				operation: ['create'],
+			},
+		},
+		routing: {
+			send: {
+				// Validate that the 'data' property is parseable as JSON and
+				// set it into the request as body.data.
+				preSend: [parseAndSetBodyJson('data', 'data')],
+			},
+		},
+		description: 'JSON data for this credential',
```

Comment:
Definitely, I will add some!

Reply:
The credentials are a bit tricky because the expected data format depends on the credential type. I'll have to leave prepopulating the available credential types as a later improvement, but it would definitely be rad.

For now, I'm adding descriptions, hints and placeholders that hopefully make figuring the expected inputs a bit easier.